#! /bin/sh /usr/share/dpatch/dpatch-run
## tigrc-parsing-fix.dpatch by Jonas Fonseca <fonseca@diku.dk>
##
## DP: Fix tokenizing when parsing ~/.tigrc
## DP:
## DP: When moving to use argv_from_string() in set_option() a regression was
## DP: introduced causing problems with parsing of multiple spaces between
## DP: arguments. For example:
## DP:
## DP: 	color date              yellow  default
## DP:
## DP: Reported by Clifford Caoile.

diff a/tig.c b/tig.c
--- a/tig.c
+++ b/tig.c
@@ -292,7 +292,7 @@ argv_from_string(const char *argv[SIZEOF_ARG], int *argc, char *cmd)
 
 		cmd[valuelen] = 0;
 		argv[(*argc)++] = chomp_string(cmd);
-		cmd += valuelen + advance;
+		cmd = chomp_string(cmd + valuelen + advance);
 	}
 
 	if (*argc < SIZEOF_ARG)
